lithium/lithium.py

```diff
@@ -55,7 +55,8 @@ def readTestcase(self, filename):
             # Determine whether the f has a DDBEGIN..DDEND section.
             for line in f:
                 if line.find(b"DDEND") != -1:
-                    raise LithiumError("The testcase (%s) has a line containing 'DDEND' without a line containing 'DDBEGIN' before it." % self.filename)
+                    raise LithiumError("The testcase ({}) has a line containing 'DDEND' ".format(self.filename) +
+                                       "without a line containing 'DDBEGIN' before it.")
```

Comment:
You can get the same behaviour by using `%s` everywhere. That's what happens when you use `{}`.  

Also I was not joking about it being slower. It's 19x slower on python3.6 and 17x slower on python2.7 for a simple example using timeit (`'%d' % 123` vs `'{}'.format(123)`)

The python community is not at consensus that this is the way to go, so I don't think we should spend time updating everything without more discussion:
- **boto**: 1234 `%` statements _vs_ 87 `.format()` statements
- **django**: 1302 _vs_ 86
- **numpy**: 1216 _vs_ 39
- **celery**: 22 _vs_ 251